spec/search-js/formatting.spec.ts

```diff
@@ -0,0 +1,88 @@
+import { getCurrencyFormatting } from "../../src/search-js/currencies"
+import { mockNostojs } from "@nosto/nosto-js/testing"
+
+const currencyFormatsMock = {
+  GBP: {
+    currencyBeforeAmount: true,
+    currencyToken: "£",
+    decimalCharacter: ".",
+    groupingSeparator: "",
+    decimalPlaces: 1
+  },
+  EUR: {
+    currencyBeforeAmount: true,
+    currencyToken: "€",
+    decimalCharacter: ",",
+    groupingSeparator: ".",
+    decimalPlaces: 0
+  },
+  USD: {
+    currencyBeforeAmount: false,
+    currencyToken: "$",
+    decimalCharacter: ".",
+    groupingSeparator: ",",
+    decimalPlaces: 2
+  },
+  INR: {
+    currencyBeforeAmount: true,
+    currencyToken: "₹",
+    decimalCharacter: ".",
+    groupingSeparator: ",",
+    decimalPlaces: 2
+  }
+}
+
+describe("currency formatting", () => {
+  it("should use currency formatting settings", () => {
+    const { formatCurrency: format } = getCurrencyFormatting({
+      defaultCurrency: "GBP",
+      currencySettings: currencyFormatsMock
+    })
+
+    expect(format(12345.0)).toEqual("£12345.0")
+    expect(format(123450.0)).toEqual("£123450.0")
+
+    expect(format(12345.0, "EUR")).toEqual("€12.345")
+    expect(format(123450.0, "EUR")).toEqual("€123.450")
+
+    expect(format(12345.0, "USD")).toEqual("12,345.00$")
+    expect(format(123450.0, "USD")).toEqual("123,450.00$")
+
+    expect(format(12345.0, "INR")).toEqual("₹12,345.00")
+    expect(format(123450.0, "INR")).toEqual("₹1,23,450.00")
+
+    // uses fallback
+    expect(format(12345.0, "AUD")).toEqual("$12,345.00")
+    expect(format(123450.0, "AUD")).toEqual("$123,450.00")
+  })
+
+  it("should have proper fallback behaviour", () => {
+    const { formatCurrency: format } = getCurrencyFormatting({
+      defaultCurrency: "GBP",
+      currencySettings: {}
+    })
+
+    expect(format(12345.0)).toEqual("£12,345.00")
+    expect(format(123450.0)).toEqual("£123,450.00")
+
+    expect(format(12345.0, "EUR")).toEqual("12.345,00 €")
+    expect(format(123450.0, "EUR")).toEqual("123.450,00 €")
+
+    expect(format(12345.0, "USD")).toEqual("$12,345.00")
+    expect(format(123450.0, "USD")).toEqual("$123,450.00")
+  })
+
+  it("should load currency settings from nosto", async () => {
+    const mockApi = {
+      internal: {
+        getSettings: jest.fn(() => ({
```

Comment:
Why is there a difference in mocking? https://github.com/Nosto/nosto-autocomplete/pull/138/files#diff-a8d872a4d919369003607c9a03c5cec47c4b2877646f212ecce08a3e2b3fc6e8R7

Reply:
mockNostojs doesn't return the mock
to be able to check the assertions there is a split into mockApi declaration and mockNostojs invociation

Reply:
Sorry for being vague in my initial comment. The point is, we mock `getSettings` to be a `jest.fn` here, but [here](https://github.com/Nosto/nosto-autocomplete/pull/138/files#diff-a8d872a4d919369003607c9a03c5cec47c4b2877646f212ecce08a3e2b3fc6e8R7) a normal function

Reply:
In case there is no need to check the mock function invocations I don't apply any jest.fn/vi.fn wrapping

Reply:
Makes sense